drivers/clock_control/clock_control_ifx_fixed_clock.c

```diff
@@ -128,20 +128,33 @@ static int fixed_rate_clk_init(const struct device *dev)
 	const struct fixed_rate_clock_config *const config = dev->config;
 
 	switch (config->system_clock) {
-
+#if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(clk_imo))
 	case IFX_IMO:
-		break;
+		Cy_SysClk_ImoEnable();
+#if defined(CONFIG_SOC_FAMILY_INFINEON_PSOC4)
```

Comment:
This should really be a hidden Kconfig informing code that PSOC4 has an adjustable clock here.